Assets/MRTK/Core/Services/BaseService.cs

```diff
@@ -101,7 +118,10 @@ public virtual bool IsMarkedDestroyed
         {
             get
             {
-                Debug.Assert(isMarkedDestroyed.HasValue, $"{GetType()} has not set a value for IsMarkedDestroyed, returning false.");
+                if (!isMarkedDestroyed.HasValue)
```

Comment:
Does this need the `if` check if it's also being converted from `Debug.Assert` to `Debug.AssertFormat` without an interpolated string? The allocation would be happening because of the interpolated string, but the method call without it will no-op instead.

Reply:
Maybe we just need to change anywhere we use interpolated strings in `Debug.Assert` calls into `Debug.AssertFormat` calls?

Reply:
Well, there's also GetType() that would still get called.  Though we also changed GetType to cache it's result so it's not as expensive, but I use it as an example of Debug.AssertFormat not necessarily being a perfect solution.

Reply:
True! A combination of caching the type _and_ using AssertFormat seems good though.

Reply:
Looked into this more...unfortunately since AssertFormat uses `params object[]` as a parameter, it allocs an array every call :( the `if` might be our only path forward @davidkline-ms

Reply:
Oh! Caching the type name as an array might get around this...digging in!

![image](https://user-images.githubusercontent.com/3580640/141357004-af1a4854-e7e3-434c-b554-95f8f1ef5816.png)

Reply:
Yeah, that dropped me to 0 allocs without the `if`: 
![image](https://user-images.githubusercontent.com/3580640/141357194-ec9b11ca-2c0a-44e3-9c1e-b9149128fab6.png)
I also refactored the strings out into `const` strings, since we're no longer interpolating them